### PR TITLE
Voice: add separate troubleshooting section

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Proposed change

Voice: add separate troubleshooting section
- In the nav aside, add separate troubleshooting section
- move page on generic Assist troubleshooting there
- add new section on troubleshooting the Espressif ESP32-S3-Box
- add redirects from `/s3-box-no-wifi` and from `/s3-box-no-ha`

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/voice_control/troubleshooting_the_s3_box.markdown [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+     2. Go to https://esphome.io/projects/index.html select the **Connect** button, then **Change Wi-Fi**.
+   - **Option 2**: You already have the ESPHome add-on installed and adopted the ESP32-S3-BOX-3 on your ESPHome dashboard.
+     1. Make sure the USB cable is plugged into the ESP32-S3-BOX-3.
+     2. In Home Assistant, go to [**Settings** > **Add-ons** > **ESPHome**](https://my.home-assistant.io/redirect/supervisor_addon/?addon=5c53de3b_esphome), and **Open Web UI**.
+     3. On the **ESP32-S3-BOX-3** add-on, select edit.
+       ![ESP32-S3-BOX-3 open config file](/images/assist/esps32-s3-edit-config.png)
+       - **Result**: An editor opens, showing the configuration file.
+       ![ESP32-S3-BOX-3 edit config file](/images/assist/esp32-edit-wifi-credentials.png)
+     4. Under the Wi-Fi component, enter your Wi-Fi credentials:
+        - Instead of `!secret wifi_ssid`, enter the name of your 2.4&nbsp;GHz Wi-Fi.
```
Comment (jlpouffier) [inline source/voice_control/troubleshooting_the_s3_box.markdown]: YOu may want to guide the users to check their secrets file before hardcoding the credential in the firmware.

It's located here:
Screenshot 2023-12-13 at 14 35 12


 It is not that hardcoding the credential is less secure.. It is just that firmware files tend to be shared, and so it is better to sanitize it.
 Also, if the user really changed the Wifi password, changing it on the secrets file will fix all ESPHome devices at once.

File: source/voice_control/troubleshooting_the_s3_box.markdown [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+This message indicates that the device could connect to the Wi-Fi, but is unable to communicate with Home Assistant.
+
+### Remedy
+
+1. If you see this message during a restart or while an update is running, wait until the restart or update is finished.
+   - In this case, there is nothing you need to do. It is expected that the device temporarily stops communicating.
+2. Make sure your device is on the same network as Home Assistant.
+   - If you have a complex network setup with VLAN, make sure it is in the same VLAN.
+3. If you see the screen below, but the ESP32-S3-BOX-3 is not listed, select **Setup another instance of ESPHome**.
```
Comment (jlpouffier) [inline source/voice_control/troubleshooting_the_s3_box.markdown]: This step feels like it is coming from nowhere.
I think we should guide users to the correct place.

Here is my proposal:

Go to Settings > Devices and Services or click here:
[Open your Home Assistant instance and show your integrations.](https://my.home-assistant.io/redirect/integrations/)

The ESP32-S3-BOX should be automatically discovered on this screen, as shown in the screenshot below
image

If it is the case: Click configure.

If it is not the case, click on ADD INTEGRATION, and search for ESPHome or click here:
[Open your Home Assistant instance and start setting up a new integration.](https://my.home-assistant.io/redirect/config_flow_start/?domain=esphome)

And manually add the IP address of the ESP32-S3-BOX, and click SUBMIT

**Review by jlpouffier — APPROVED** [review]
All good!!

File: source/voice_control/troubleshooting_the_s3_box.markdown [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+1. First, check if your network is ready in general.
+   1. Make sure your router is on and within reach.
+   2. Make sure you have chosen a Wi-Fi network that supports 2.4 GHz. The ESP32-S3-BOX-3 won't show up on a 5 GHz network.
+
+2. The next step is to make sure you entered the correct Wi-Fi password. Follow the steps either under **Option 1** or **Option 2**, depending on whether or not you have the ESPHome add-on installed.
+   - **Option 1**: You do not have the ESPHome add-on installed or you have the add-on but did **not** adopt the ESP32-S3-BOX-3. If the device is shown in green, it is not adopted. 
+      ![ESP32-S3-BOX-3 not adopted](/images/assist/esp32-not-adopted.png)
+     1. Make sure the USB cable is plugged into the ESP32-S3-BOX-3.
+     2. Go to https://esphome.io/projects/index.html select the **Connect** button, then **Change Wi-Fi**.
```
Comment (frenck) [inline source/voice_control/troubleshooting_the_s3_box.markdown]: Do we want this as a raw URL?

**Review by frenck — APPROVED** [review]
Thanks, @c0ffeeca7 👍 

../Frenck

**Pull request merged**